File: examples/cxx/test_functions/models/eggholder.cpp
#include "eggholder.h"
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

// Minus Eggholder
void m_eggholder(std::vector<double>& pars, std::vector<double>& result)
{
 size_t nPars = pars.size();
 if( nPars != 2 ) { printf("eggholder: error, num pars must be 2 (is %zu)\n", nPars); exit(-1); }

 double s = -(pars[1]+47.0)*sin(sqrt(abs(0.5*pars[0]+pars[1]+47.0))) - pars[0]*sin(sqrt(abs(pars[0]-pars[1]-47.0)));

 result.push_back(-s);
}

File: examples/python/test_functions/model/eggholder.py
#!/usr/bin/env python
import math

# Minus Eggholder
def m_eggholder(x):
    par1 = x.getParameter(0)
    par2 = x.getParameter(1)

    result = s = -(par2+47.0)*math.sin(math.sqrt(abs(0.5*par1+par2+47.0))) - par1*math.sin(math.sqrt(abs(par1-par2-47.0)));
    
    x.addResult(-result)
